scilpy/tracking/tools.py

```diff
@@ -1,130 +1,189 @@
 
+    return subset_sft
 
-def resample_streamlines(streamlines, num_points=0, arc_length=False):
+
+def resample_streamlines_num_points(sft, num_points):
     """
     Resample streamlines using number of points per streamline
 
     Parameters
     ----------
-    streamlines: list
-        List of list of 3D points.
+    sft: StatefulTractogram
+        SFT containing the streamlines to subsample.
     num_points: int
         Number of points per streamline in the output.
-    arc_length: bool
-        Whether to downsample using arc length parametrization.
 
     Return
     ------
-    resampled_streamlines: list
-        List of resampled streamlines.
+    resampled_sft: StatefulTractogram
+        The resampled streamlines as a sft.
     """
+
+    # Checks
+    if num_points <= 1:
+        raise ValueError("The value of num_points should be greater than 1!")
+
+    # Resampling
     resampled_streamlines = []
-    for streamline in streamlines:
-        if arc_length:
-            line = set_number_of_points(streamline, num_points)
-        else:
-            line = downsample(streamline, num_points)
+    for streamline in sft.streamlines:
+        line = set_number_of_points(streamline, num_points)
         resampled_streamlines.append(line)
 
-    return resampled_streamlines
+    # Creating sft
+    resampled_sft = _warn_and_save(resampled_streamlines, sft)
+
+    return resampled_sft
+
+
+def resample_streamlines_step_size(sft, step_size):
+    """
+    Resample streamlines using a fixed step size.
+
+    Parameters
+    ----------
+    sft: StatefulTractogram
+        SFT containing the streamlines to subsample.
+    step_size: float
+        Size of the new steps, in mm.
+
+    Return
+    ------
+    resampled_sft: StatefulTractogram
+        The resampled streamlines as a sft.
+    """
+
+    # Checks
+    if step_size == 0:
+        raise ValueError("Step size can't be 0!")
+    elif step_size < 0.1:
+        logging.debug("The value of your step size seems suspiciously low. "
+                      "Please check.")
+    elif step_size > np.max(sft.voxel_sizes):
+        logging.debug("The value of your step size seems suspiciously high. "
+                      "Please check.")
+
+    # Make sure we are in world space
+    orig_space = sft.space
+    sft.to_rasmm()
+
+    # Resampling
+    lengths = length(sft.streamlines)
+    nb_points = np.ceil(lengths / step_size).astype(int)
+    if np.any(nb_points == 1):
+        logging.warning("Some streamlines are shorter than the provided "
+                        "step size...")
+        nb_points[nb_points == 1] = 2
+    resampled_streamlines = [set_number_of_points(s, n) for s, n in
+                             zip(sft.streamlines, nb_points)]
+
+    # Creating sft
+    resampled_sft = _warn_and_save(resampled_streamlines, sft)
+
+    # Return to original space
+    resampled_sft.to_space(orig_space)
+
+    return resampled_sft
+
+
+def _warn_and_save(new_streamlines, sft):
```

Comment:
It's just because I had some comment that it would be nice to really show that resample_num_points and resample_step_size are very similar. 

This part of code is the same for both.